Dockerfile

```diff
@@ -26,7 +26,8 @@ RUN python3 -m venv $VIRTUAL_ENV
 # in ENTRYPOINT or CMD.
 ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 
-RUN pip install --upgrade --no-cache-dir "cython<3" pip wheel
```

Comment:
Rationale:
- We needed to pin cython back right after the cython 3.0.0 release because some dependencies had not caught up. This is no longer needed.
- I used to think we should upgrade to latest `pip` and `wheel`, but now I think it's better take whatever the base image gives us. Upgrading was running into weird issues, and I think it's a sign that I made the wrong choice.